store/postgres/template.go

```diff
@@ -33,28 +33,33 @@ func (r TemplateRepository) Migrate() error {
 	return nil
 }
 
-func (r TemplateRepository) Upsert(template *model.Template) (*model.Template, error) {
+func (r TemplateRepository) Upsert(template *domain.Template) (*domain.Template, error) {
 	var newTemplate, existingTemplate model.Template
-	result := r.db.Where(fmt.Sprintf("name = '%s'", template.Name)).Find(&existingTemplate)
+	modelTemplate := &model.Template{}
+	_, err := modelTemplate.FromDomain(template)
```

Comment:
let's only return `error` from `FromDomain` since we already pass the param as the reference